app/controllers/tours_controller.rb

```diff
@@ -1,6 +1,27 @@
 class ToursController < ApplicationController
+  # before_action :search_upcoming, :filter_tours, only: :index
+
   def index
-    @pagy, @tours = pagy(Tour.upcoming,
-                         items: Settings.tour.items_per_page)
+    @tours = Tour.all
+    @tour_types = TourType.all
+    search
+    @pagy, @tours = pagy(@tours, items: Settings.tour.items_per_page)
+  end
+
+  private
+
+  def search
+    Tour::SEARCH_ATTRIBUTES.each do |field|
+      @tours = @tours.send "by_#{field}", params[field]
+    end
+    if params[:or]
+      # rubocop:disable Layout/LineLength
+      @tours = @tours.by_city(params[:city]).or(@tours.by_tour_type_id(params[:tour_type_id]))
```

Comment:
```suggestion
      @tours = @tours.by_city(params[:city])
                      .or(@tours.by_tour_type_id(params[:tour_type_id]))
```
Thử như này xem code có chạy đc ko nèo